cmd/format-xl.go

```diff
@@ -753,16 +754,43 @@ func fixFormatXLV3(storageDisks []StorageAPI, endpoints Endpoints, formats []*fo
 func initFormatXL(ctx context.Context, storageDisks []StorageAPI, setCount, drivesPerSet int, deploymentID string) (*formatXLV3, error) {
 	format := newFormatXLV3(setCount, drivesPerSet)
 	formats := make([]*formatXLV3, len(storageDisks))
-
+	logger.Info("Formatting Zone...")
 	for i := 0; i < setCount; i++ {
+		logger.Info(" * Set %v:", i+1)
+		hostCount := make(map[string]int, drivesPerSet)
 		for j := 0; j < drivesPerSet; j++ {
+			disk := storageDisks[i*drivesPerSet+j]
 			newFormat := format.Clone()
 			newFormat.XL.This = format.XL.Sets[i][j]
 			if deploymentID != "" {
 				newFormat.ID = deploymentID
 			}
+			logger.Info("   - Drive: %s", disk.String())
+			hostCount[disk.Hostname()]++
 			formats[i*drivesPerSet+j] = newFormat
 		}
+		if len(hostCount) > 1 {
+			// Config
+			wantAtMost := globalStorageClass.GetParityForSC(storageclass.STANDARD)
+			if wantAtMost == 0 {
```

Comment:
Can this become a function of its own @klauspost ? Also looks like it doesn't require repeated calculation because drivesPerSet is same across zones as we keep common SLA for all zones.